202305_coastal_ecosystems_TC/impact_calculation/exposure_ecosystem_matching.py

```diff
@@ -0,0 +1,101 @@
+"""
+This file is part of CLIMADA.
+
+Copyright (C) 2017 ETH Zurich, CLIMADA contributors listed in AUTHORS.
+
+CLIMADA is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free
+Software Foundation, version 3.
+
+CLIMADA is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along
+with CLIMADA. If not, see <https://www.gnu.org/licenses/>.
+
+---
+
+Script to create population exposure and match with habitat protective capacity data
+Author: Sarah Hülsen
+"""
+
+import geopandas as gpd
+from climada.entity import Exposures
+import pandas as pd
+
+# This script creates an exposure file from vectorised WorldPop population data (per TC basin)
+# It also merges habitat data with the population data (4 permutations of two habitat datasets and two pop datasets)
+# The merged files can later be used to analyse impacts per habitat category
+
+# variables
+geo_epsg = 4326  # WGS84
+proj_eck4 = '+proj=eck4 +lon_0=0 +x_0=0 +y_0=0 +datum=WGS84 +units=m +no_defs'  # World Eckert IV
+prot_dist = 2000
+pop_years = ['2000', '2020']
+hab_years = ['1992', '2020']
+basins = ['AP', 'WP', 'IO', 'SH']
+input_path = '/Users/USERNAME/Documents/CLIMADA/coastal_ecosystems/data/'   # path to input data 
+output_path = '/Users/USERNAME/Documents/CLIMADA/coastal_ecosystems/results/intermediate/'  # path to save results
```

Comment:
Why put such a Path? This is specific to the MacOS. The trick is to use relative paths.

```suggestion
input_path = '../data/'   # path to input data 
output_path = '../'  # path to save results
```

Then add a folder `impact_calculation/data/`.

Reply:
Also, in if you want to make it very simple for cross-platform users, you can use the Pathlib library instead of using strings to define paths.